library/model/book.py

```diff
@@ -96,8 +62,10 @@ def borrow_book(self) -> "BorrowedBook":
 
     def __eq__(self, other):
         """Overrides the default implementation"""
-        if isinstance(other, Book) or isinstance(other, BorrowedBook):
```

Comment:
you don't need to do both checks: BorrowedBook inherits from book, so every BorrowedBook automatically is a Book already.
Did you mean `type(self) is BorrowedBook`?